### PR TITLE
Add validation that cargo.toml version matches most recent CHANGELOG.md version.

### PR DESCRIPTION
**Pull request by exzachlyvv**

### CODE REVIEW

File: crates/cli-tools/src/changelog.rs [hunk 1]
```diff
@@ -173,6 +173,21 @@ impl Changelog {
 
         Ok(changelog)
     }
+
+    fn validate_cargo_toml(&self, path: &str) -> Result<()> {
+        let most_recent_release =
+            self.releases.first().context("At least 1 release is required")?;
```
Comment (ia0) [inline crates/cli-tools/src/changelog.rs:179]: Can this happen? We only produce valid changelogs right? So it's an invariant that `releases` is not empty.

File: crates/cli-tools/src/changelog.rs [hunk 2]
```diff
@@ -228,7 +243,11 @@ pub async fn execute_ci() -> Result<()> {
         let path = path?;
 
         // Validation done during parsing.
-        Changelog::read_file(&path).await?;
+        let changelog = Changelog::read_file(&path).await?;
+
+        let crate_dir = path.strip_suffix("/CHANGELOG.md").context("Malformed file path")?;
```
Comment (ia0) [inline crates/cli-tools/src/changelog.rs:248]: Same, I think we can `unwrap()` here. This is not a user error, this is an internal bug.

**Review by ia0 — COMMENTED** [review]
Thanks! LGTM

File: crates/cli-tools/src/changelog.rs (hunk 1 above)
Comment (exzachlyvv) [reply crates/cli-tools/src/changelog.rs:179]: Correct, that cannot happen. I'll use `unwrap()` instead.

**Review by ia0 — APPROVED** [review]

**Pull request merged**